### PR TITLE
Send analytics to FPTI

### PR DESCRIPTION
**Pull request by scannillo**
### Summary of changes

- Add `FPTIBatchData`
    - Encapsulates & formats JSON post of batch event data to FPTI
    - Populates static device-specific properties (ex: isSimulator)
- Update `BTAnalyticsService`
    - Swap FPTI url for previous analytics URL parsed from config
    - Use existing logic to construct batches of events per session
    - Add docstrings to existing methods for clarity
- `BTHTTP`
    - Add `post()` method that accepts `Encodable` parameters
        - For now, this new `post()` method will just call internally to the one that sends a raw dictionary type, but this sets the stage for us to begin converting all of our POST params to `Encodable` structs
    - Prevent `BTHTTP` from automatically tacking on an `"authorization_fingerprint"` key-value pair into the POST body for requests hitting `paypal.com` URLs
        - (I honestly never realized we were doing this. Created _DTBTSDK-2682_ to investigate across JS and Android)

### Checklist

- [X] Added a changelog entry
- Added and linked JIRAs for tasks to be done before GA release

### Authors
@scannillo 

### Verification

I verified using Looker Event Explorer that our events (and metadata/tags) are successfully being sent to FPTI:
bt_fpti

### CODE REVIEW

File: Sources/BraintreeCore/Analytics/BTAnalyticsMetadata.swift [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import UIKit
 
+// TODO: - Update to mirror params sent in Android and JS SDKs. Currently these analytics details params are being sent in all POST bodies.
+// JIRA - DTBTSDK-2683
```
Comment (scannillo) [inline Sources/BraintreeCore/Analytics/BTAnalyticsMetadata.swift:4]: Addressed in https://github.com/braintree/braintree_ios/pull/1024

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ## unreleased
 * Require Xcode 14.1 (per [App Store requirements](https://developer.apple.com/news/?id=jd9wcyov#:~:text=Starting%20April%2025%2C%202023%2C%20iOS,on%20the%20Mac%20App%20Store))
+* Send analytic events to FPTI instead of Arachne
```
Comment (hollabaq86) [inline CHANGELOG.md:5]:
```suggestion
* Change analytics endpoint
```

Comment (hollabaq86) [reply CHANGELOG.md:5]: I don't think we need to publicly mention service names that we're changing to, what do you think?

**Review by hollabaq86 — COMMENTED** [review]
end of an era! small Q

File: CHANGELOG.md (hunk 2 above)
Comment (scannillo) [reply CHANGELOG.md:5]: Good question - I was just thinking of what would be most clear for our future-selves when we want to hunt down when we made the switch.

Maybe we can be more generic like `Move to PayPal analytics service`?

Comment (hollabaq86) [reply CHANGELOG.md:5]: Yeah I think that’s a great compromise! 🚢

File: Sources/BraintreeCore/Analytics/FPTIBatchData.swift [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import Foundation
+import UIKit
+
+/// The POST body for a batch upload of FPTI events
+struct FPTIBatchData: Codable {
+    let events: [EventsContainer] // Single-element "events" array required by FPTI formatting
```
[the remark this answers is not in the document]
Comment (scannillo) [reply Sources/BraintreeCore/Analytics/FPTIBatchData.swift:6]: Newline? Will do!

File: CHANGELOG.md (hunk 2 above)
Comment (scannillo) [reply CHANGELOG.md]: Done!

File: Sources/BraintreeCore/Analytics/BTAnalyticsService.swift [hunk 4]
```diff
@@ -73,13 +55,15 @@ class BTAnalyticsService: Equatable {
         }
     }
 
+    /// Sends request to FPTI immediately, without checking # of events in queue against flush threshold
```
[the remark this answers is not in the document]
Comment (scannillo) [reply Sources/BraintreeCore/Analytics/BTAnalyticsService.swift]: Done!

File: Sources/BraintreeCore/URL+IsPayPal.swift [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import Foundation
+
+extension URL {
+    
+    /// Used to determine if a the URL is a paypal.com domain in order to format API requests accordingly
+    var isPayPalURL: Bool {
+        return self.absoluteString.contains("api-m.paypal.com")
```
[the remark this answers is not in the document]
Comment (scannillo) [reply Sources/BraintreeCore/URL+IsPayPal.swift]: Done!

File: Sources/BraintreeCore/Analytics/FPTIBatchData.swift [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    
+    init(metadata: Metadata, events fptiEvents: [Event]?) {
+        self.events = [EventsContainer(
+            metadata: metadata,
+            fptiEvents: fptiEvents ?? []
+        )]
+    }
```
[the remark this answers is not in the document]
Comment (jaxdesmarais) [reply Sources/BraintreeCore/Analytics/FPTIBatchData.swift:14]: Ah dang. That makes sense though. Thanks for giving it a try and letting me know the results!

**Review by jaxdesmarais — APPROVED** [review]

**Pull request merged**